source/Halibut/Transport/Protocol/ITypeRegistry.cs

```diff
@@ -0,0 +1,12 @@
+﻿using System;
+
+namespace Halibut.Transport.Protocol
+{
+    public interface ITypeRegistry
+    {
+        void Register(params Type[] registeredServiceTypes);
```

Comment:
The 3 methods `Register`, `RegisterType`, and `AddToMessageContract` are all fancy ways of registering a type.

It's a little unclear what each is for looking at the interface. Except maybe `RegisterType`

A suggestion is to maybe rename `Register` to something like `RegisterTypesFromService`?